clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -40,15 +48,28 @@ public KafkaMetric(Object lock, MetricName metricName, MetricValueProvider<?> va
         this.time = time;
     }
 
+    /**
+     * Get the configuration of this metric
+     * @return Return the config of this metric
+     */
     public MetricConfig config() {
         return this.config;
```

Comment:
this is unrelate to this PR, but should we add lock to this method? It seems we assume the config can be read/write concurrently.

Reply:
I think you are right, there is a lock for setter, we should have a lock for getter as well. 
Thanks for the suggestion, let me add that